### PR TITLE
Require user-name, user-email, branch and use original commit message by default

### PR DESCRIPTION
**Pull request by ruudk**
See commits.

### CODE REVIEW

File: packages/monorepo-split-github-action/entrypoint.sh [hunk 1]
```diff
@@ -50,6 +51,14 @@ note "Cleaning destination repository of old files"
 find "$CLONE_DIR" | grep -v "^$CLONE_DIR/\.git" | grep -v "^$CLONE_DIR$" | xargs rm -rf # delete all files (to handle deletions)
 ls -la "$CLONE_DIR"
 
+if test ! -z "$COMMIT_MESSAGE"
+then
+    ORIGIN_COMMIT="https://github.com/$GITHUB_REPOSITORY/commit/$GITHUB_SHA"
+    COMMIT_MESSAGE="${COMMIT_MESSAGE/ORIGIN_COMMIT/$ORIGIN_COMMIT}"
+else
+    COMMIT_MESSAGE=$(git show -s --format=%B "$GITHUB_SHA")
```
Comment (ruudk) [inline packages/monorepo-split-github-action/entrypoint.sh:59]: @TomasVotruba What do you think? I'm using this trick for [LongRunning](https://github.com/LongRunning/LongRunning). 

See mono commit https://github.com/LongRunning/LongRunning/commit/62eae751184e87506870c096e76084b53cbb0fdb
and split commit https://github.com/LongRunning/core/commit/a8b565172fccda4cbef1dbaf52eda2fbfa0dd07a

File: packages/monorepo-split-github-action/action.yaml [hunk 2]
```diff
@@ -10,21 +10,24 @@ inputs:
         description: 'Remote organization'
         required: true
     split-repository-name:
-        description: 'Remove repository'
+        description: 'Remote repository'
         required: true
     commit-message:
-        description: '[Optional] commit message for the output repository. ORIGIN_COMMIT is replaced by the URL@commit in the origin repo'
-        default: 'Package split ORIGIN_COMMIT'
+        description: '[Optional] commit message for the output repository. ORIGIN_COMMIT is replaced by the URL@commit in the origin repo.'
```
Comment (TomasVotruba) [inline packages/monorepo-split-github-action/action.yaml:16]: 👍

File: packages/monorepo-split-github-action/action.yaml [hunk 3]
```diff
@@ -10,21 +10,24 @@ inputs:
...
         required: false
+    branch:
+        description: 'Git branch'
+        default: 'main'
+        required: true
```
Comment (TomasVotruba) [inline packages/monorepo-split-github-action/action.yaml:21]: Do you know any simple way to switch all `master` branches to `main`?

I'd have to update all the projects now.
Maybe go with `master` now and handle rename in new PR

File: packages/monorepo-split-github-action/entrypoint.sh (hunk 1 above)
Comment (TomasVotruba) [reply packages/monorepo-split-github-action/entrypoint.sh:59]: Cool, let's try it

File: packages/monorepo-split-github-action/entrypoint.sh [hunk 4]
```diff
@@ -50,17 +51,22 @@ note "Cleaning destination repository of old files"
...
+fi
+
 note "Copying contents to git repo"
 
-cp -r "$PACKAGE_DIRECTORY"/* "$CLONE_DIR"
+cp -Ra $PACKAGE_DIRECTORY/. "$CLONE_DIR"
```
Comment (TomasVotruba) [inline packages/monorepo-split-github-action/entrypoint.sh:66]: What is this for?

Comment (TomasVotruba) [reply packages/monorepo-split-github-action/entrypoint.sh:66]: Comment is needed so the knowledge is in the code for future readers

File: packages/monorepo-split-github-action/action.yaml (hunk 3 above)
Comment (ruudk) [reply packages/monorepo-split-github-action/action.yaml:21]: Switched to master.

File: packages/monorepo-split-github-action/entrypoint.sh (hunk 4 above)
Comment (ruudk) [reply packages/monorepo-split-github-action/entrypoint.sh:66]: Added.

**Comment by ruudk** [comment]
@TomasVotruba Updated the PR.

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you

**Comment by ruudk** [comment]
@TomasVotruba I see that the split fails, is that expected? https://github.com/symplify/github-actions/runs/1486094306?check_suite_focus=true

**Comment by TomasVotruba** [comment]
It should pass as before

**Comment by TomasVotruba** [comment]
Oh, it was triggered by your account, that does not have the token. That's weird.

image


I'll look into it.

**Comment by TomasVotruba** [comment]
So it needed to add "workflow" scope to the token:

image

The split works now :+1:

**Pull request merged**